beacon_node/http_api/src/lib.rs

```diff
@@ -3184,29 +3245,45 @@ pub fn serve<T: BeaconChainTypes>(
         .and(task_spawner_filter.clone())
         .and(chain_filter.clone())
         .then(
-            |query: api_types::ValidatorAggregateAttestationQuery,
+            |endpoint_version: EndpointVersion,
+             query: api_types::ValidatorAggregateAttestationQuery,
              not_synced_filter: Result<(), Rejection>,
              task_spawner: TaskSpawner<T::EthSpec>,
              chain: Arc<BeaconChain<T>>| {
                 task_spawner.blocking_json_task(Priority::P0, move || {
                     not_synced_filter?;
-                    chain
-                        .get_pre_electra_aggregated_attestation_by_slot_and_root(
+                    let res = if endpoint_version == V2 {
```

Comment:
should `match`?

Reply:
`V2` is just a struct wrapper around a `u64` not an enum

```
pub const V2: EndpointVersion = EndpointVersion(2);

#[derive(Debug, Clone, Copy, PartialEq)]
pub struct EndpointVersion(pub u64);
```